src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -648,12 +669,76 @@ public void RemoveAnalyzerReference(string fullPath)
 
         #endregion
 
-        private void FileChangeContext_FileChanged(object sender, string fullFilePath)
+        private void DocumentFileChangeContext_FileChanged(object sender, string fullFilePath)
         {
             _sourceFiles.ProcessFileChange(fullFilePath);
             _additionalFiles.ProcessFileChange(fullFilePath);
         }
 
+        #region Metadata Reference Refreshing
+
+        private void FileReferenceChangeContext_FileChanged(object sender, string fullFilePath)
+        {
+            lock (_gate)
+            {
+                CancelOutstandingMetadataReferenceRefreshForFile_NoLock(fullFilePath);
+
+                var cancellationTokenSource = new CancellationTokenSource();
+                _metadataReferenceRefreshCancellationTokenSources.Add(fullFilePath, cancellationTokenSource);
+
+                Task.Delay(TimeSpan.FromSeconds(5), cancellationTokenSource.Token).ContinueWith(_ =>
```

Comment:
so we no longer use SafeContinueWith thing? asking since one good thing about that is it catch exception. I guess, this will swallow the exception?

Reply:
I mostly had forgotten. In the case of it not refreshing, that's probably something we want though...

Reply:
The same delay was previously hardcoded: https://github.com/dotnet/roslyn/blob/5545eebd717354440a54c5a22d284d85d18a752d/src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs#L683